pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -740,4 +743,21 @@ protected AsyncLoadingCache<NamespaceName, SystemTopicClient.Writer<PulsarEvent>
     }
 
     private static final Logger log = LoggerFactory.getLogger(SystemTopicBasedTopicPoliciesService.class);
+
+    @Override
+    public void close() throws Exception {
+        writerCaches.synchronous().invalidateAll();
+        readerCaches.values().forEach(future -> {
+            if (future != null && !future.isCompletedExceptionally()) {
+                future.thenAccept(reader -> {
+                    try {
+                        reader.close();
+                    } catch (IOException e) {
```

Comment:
Maybe catching `Exception` might be safer?

Reply:
done